packages/mdc-floating-label/foundation.ts

```diff
@@ -31,25 +31,11 @@ export class MDCFloatingLabelFoundation extends MDCFoundation<MDCFloatingLabelAd
     return cssClasses;
   }
 
-  /**
-   * See {@link MDCFloatingLabelAdapter} for typing information on parameters and return types.
-   */
-  static get defaultAdapter(): MDCFloatingLabelAdapter {
-    // tslint:disable:object-literal-sort-keys Methods should be in the same order as the adapter interface.
-    return {
-      addClass: () => undefined,
-      removeClass: () => undefined,
-      getWidth: () => 0,
-      registerInteractionHandler: () => undefined,
-      deregisterInteractionHandler: () => undefined,
-    };
-    // tslint:enable:object-literal-sort-keys
-  }
 
   private readonly shakeAnimationEndHandler_: SpecificEventListener<'animationend'>;
 
-  constructor(adapter?: Partial<MDCFloatingLabelAdapter>) {
-    super({...MDCFloatingLabelFoundation.defaultAdapter, ...adapter});
+  constructor(protected readonly adapter: MDCFloatingLabelAdapter) {
```

Comment:
You can just do (adaptor: MDCFloatingLabelAdapter) since the super(adaptor) call will take care of adding it as a protected readonly member